modules/unicode.js

```diff
@@ -0,0 +1,30 @@
+var request = require('request-promise')
+module.exports = {
+  commands: {
+    unicode: {
+      aliases: [ 'char' ],
+      help: 'Returns a unicode character whose name or codepoint matches a regular expression',
+      command: async function (bot, msg) {
+        if (msg.args.length === 0) {
+          // This will work when (if) #162 is merged
+          msg.body = '.*'
+        }
+        try {
+          const response = await request.get('https://unicode.org/Public/UNIDATA/UnicodeData.txt')
+          const data = response.split('\n')
+          const regex = new RegExp(msg.body, 'i')
+          const matches = data.filter(codepoint => codepoint.match(regex) &&
+            codepoint.match(/;(L[lmtuo]|M[cen]|N[dlo]|P[cdifseo]|S[ckmo]|Zs);/))
```

Comment:
I'm scared of this regex

Reply:
It's not too scary really, it's just filtering by [character category](https://unicodebook.readthedocs.io/unicode.html#categories), to allow:
* Letters
* Marks
* Numbers
* Punctuation
* Symbols
* *Space* separators (but not paragraph or line separators)

All control characters are excluded.

`;([LMNPS].|Zs);` would be cleaner, but I thought it may be worthwile to explicitly list the allowed categories.

The two letter category codes never constitute a semicolon-delimited field of their own with any meaning other than the character category.

Reply:
Would it be cleaner to blacklist instead of whitelist? Do we only care if a character isn't `;(C|Z).;`?

Also, it's fine if we talk about these scary characters, as long as we don't output them.
At the moment, we'd ask the bot for line feed characters and it'll tell us that there aren't any. What it should do is tell you about them, but not actually output the character itself.

Perhaps we could simply filter whether the character is printable? ([this npm module perhaps](https://www.npmjs.com/package/printable-characters) -- I've not looked at it closely)

Reply:
Filtering for *not* `;(C.|Z[lp]);` would work, as long as we're not concerned about the Unicode consortium adding more categories (which is, admittedly, unlikely).

Reply:
377a791